components/template-variables.tpl

```diff
@@ -342,6 +342,43 @@
     {% assign body_bg_color_data_str = body_bg_color_data | json %}
   {% endif %}
 
+  {% comment %}AUTO-RENDERED PRODUCT PAGE BACKGROUND VARIABLES{% endcomment %}
+  {% assign product_bg_key = template_settings.site.product_settings.key %}
+
+  {% assign product_bg = site.data[product_bg_key] %}
+  {% assign product_bg_color = product_bg.color %}
+  {% assign product_bg_color_data = product_bg.colorData %}
+  {% assign product_bg_combined_lightness = product_bg.combinedLightness %}
+
+  {% if product_bg %}
```

Comment:
It's better to use `product_bg != blank`. It deals with empty string, nil, empy hahs etc. Current code deals with only `nil` and `false`  values.

```suggestion
  {% if product_bg != blank %}
```